pkg/controller/configmap/configmap.go

```diff
@@ -84,6 +84,10 @@ func (r *ReconcileConfigMap) syncConfigMap(
 			for {
 				updateMetaGenerator := &kubecluster
 				annotations := updateMetaGenerator.Annotations
+				// if annotations == nil {
+				// 	annotations := make(map[string]string)
+				// 	updateMetaGenerator.Annotations = annotations
+				// }
```

Comment:
What's up here?

Reply:
I am still trying to reproduce an issue I saw just once, my guess is - while the cluster is being deleted the cluster obj is there but annotation has been removed if that happens it might panic below. Not sure. Do you see a race window here?

Reply:
Eh regardless, this block of code should not assume & depend on annotations being non-nil. It's good to have the check before we start trying to deref it.